number_parser/parser.py

```diff
@@ -153,41 +153,61 @@ def _normalize_tokens(token_list):
     return [_strip_accents(token.lower()) for token in token_list]
 
 
-def _normalize_dict(lang_dict):
+def _normalize_dict(lang_data):
     """Removes the accent from each key of input dictionary"""
-    return {_strip_accents(word): number for word, number in lang_dict.items()}
+    return {_strip_accents(word): number for word, number in lang_data.items()}
 
 
-def _is_cardinal_token(token, lang_dict):
-    if token in lang_dict.all_numbers:
+def _is_cardinal_token(token, lang_data):
+    """Checks if the given token is a cardinal number and returns token"""
+    if token in lang_data.all_numbers:
         return token
     return None
 
 
-def _is_ordinal_token(token, lang_dict):
-    token = _apply_cardinal_conversion(token, lang_dict)
-    return _is_cardinal_token(token, lang_dict)
+def _is_ordinal_token(token, lang_data):
+    """Checks if the given token is a ordinal number and returns token"""
+    if _is_cardinal_token(token, lang_data) is None:
+        return _is_number_token(token, lang_data)
+    return None
 
 
-def _is_number_token(token, lang_dict):
-    return _is_cardinal_token(token, lang_dict) or _is_ordinal_token(token, lang_dict)
+def _is_number_token(token, lang_data):
+    """
+    Checks if the given token belongs to either cardinal or ordinal numbers
+    and returns the cardinal form.
+    """
+    token = _apply_cardinal_conversion(token, lang_data)
+    return _is_cardinal_token(token, lang_data)
 
 
-def _apply_cardinal_conversion(input_string, language):
-    # this will be coming from the LanguageData
+def _apply_cardinal_conversion(token, lang_data):
```

Comment:
The function is updated so that now it handles each token rather than an entire string. Now when it  is used by ```parse``` with words ending in ``th`` that are not numbers like  ```with , goldsmith``` these are not changed.